scripts/Muon/GUI/ElementalAnalysis2/load_widget/load_utils_ea.py

```diff
@@ -0,0 +1,77 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from mantid.api import WorkspaceGroup
+from mantid.simpleapi import DeleteWorkspace
+
+
+def combine_loaded_runs(model, run_list):
+    """
+        As a result of this function there should be:-
+            - a groupworkspace named after the range of runs used e.g. '1234-1237'
+            - a workspace for each detector named [detector]_[groupworkspace]
+                e.g. 'Detector 1_1234-1237'
+    """
+    co_add_workspace_name = str(run_list[0]) + "-" + str(run_list[-1])
+    co_add_workspace = WorkspaceGroup()
+    finished_detectors = []
+    ws_remove = []
+    for run in run_list:
+        run_detectors = get_detectors(model, run)
+        detectors_to_check = check_all_detectors_complete(run_detectors, finished_detectors)
+        if detectors_to_check:
+            for detector in run_detectors:
+                new_ws_name = detector
+                new_ws = find_ws_to_use(model, run_detectors, detector, run)
+                for current_run in run_list:
+                    if current_run != run:
+                        all_detectors = get_detectors(model, current_run)
+                        if detector in all_detectors:
+                            workspace_to_add = find_ws_to_use(model, run_detectors, detector, current_run)
+                            new_ws = new_ws + workspace_to_add
+                add_detector_workspace_to_group(co_add_workspace, new_ws, new_ws_name, detector, finished_detectors,
+                                                ws_remove)
+    finalise_groupworkspace(co_add_workspace, co_add_workspace_name, ws_remove)
+
+
+def check_all_detectors_complete(run_detectors, finished_detectors):
+    unused_detectors = list(set(run_detectors).difference(finished_detectors))
```

Comment:
could we not just return this line and change it to check_for_unusued_detectros?